assets/js/translator.js

```diff
@@ -741,6 +764,125 @@ function translateDoc() {
+    // insert noise using document.write that 1. makes things enormously slow, and 2. completely mess up styling so e.g. you
+    // have to scroll through a full screen of whitespace before reaching content. This might mess things up some places – needs
+    // testing – but on the other hand most uses of document.write are evil.
+    return html.replace(/document[.]write[(]/g, 'console.log("document.write "+');
+}
+
+function translateWebpage() {
+    persistChoices('translator', true);
+    if(!$('div#translateWebpage').is(':visible')) {
+        showTranslateWebpageInterface($('#originalText').val().trim());
+    }
+
+    if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
+        sendEvent('translator', 'translateWebpage', curSrcLang + '-' + curDstLang);
+        if(textTranslateRequest) {
+            textTranslateRequest.abort();
+            clearTimeout(apyRequestTimeout);
+        }
+        $('iframe#translatedWebpage').animate({'opacity': 0.75}, 'fast');
+        textTranslateRequest = callApy({
+            data: {
+                'langpair': curSrcLang + '|' + curDstLang,
+                'markUnknown': 'no',
+                'url': $('input#webpage').val()
+            },
+            dataType: 'json',
+            success: handleTranslateWebpageSuccessResponse,
+            error: handleTranslateWebpageErrorResponse,
+            complete: function () {
+                ajaxComplete();
+                textTranslateRequest = undefined;
+                $('iframe#translatedWebpage').animate({'opacity': 1}, 'fast');
+            }
+        }, '/translatePage', true);
+    }
+}
+
+function handleTranslateWebpageSuccessResponse(data) {
+    if(data.responseStatus === HTTP_OK_CODE) {
+        var iframe = $('<iframe id="translatedWebpage" class="translatedWebpage" frameborder="0"></iframe>')[0];
+        $('#translatedWebpage').replaceWith(iframe);
+        iframe.contentWindow.document.open();
+        var html = cleanPage(data.responseData.translatedText);
+        iframe.contentWindow.document.write(html);
+        iframe.contentWindow.document.close();
+        var contents = $(iframe).contents();
+        contents.find('head')
+            .append($('<base>').attr('href', $('input#webpage').val()));
+        $(iframe).load(function () {
+            $.each(contents.find('a'), function (index, a) {
+                var href = a.href;
+                $(a).on('click', function () { window.parent.translateLink(href); });
+                a.href = '#';
+                a.target = '';
+            });
+        });
+    }
+    else {
+        translationNotAvailableWebpage(data);
+    }
+}
+
+function handleTranslateWebpageErrorResponse(jqXHR) {
+    translationNotAvailableWebpage(jqXHR.responseJSON);
+}
+
+function showTranslateWebpageInterface(url) {
+    $('#srcLangSelectors').css({
```

Comment:
Um, here `#srcLangSelectors` has a CSS in style.css file. Thus, even when I add a separate class to provide the styling of the hardcoded numbers, it does not let go of the original styling (It just adds to the original styling!) Would it make sense to make 2 separate classes holding the following for srcLangSelectors:
1. the style originally held
2. the style on webpage translation
And remove the style held for `#srcLangSelectors` css?